spring-integration-core/src/main/java/org/springframework/integration/dsl/context/DslIntegrationConfigurationInitializer.java

```diff
@@ -58,11 +59,15 @@ public void initialize(ConfigurableListableBeanFactory configurableListableBeanF
 		);
 
 		BeanDefinitionRegistry registry = (BeanDefinitionRegistry) configurableListableBeanFactory;
+
 		if (!registry.containsBeanDefinition(INTEGRATION_FLOW_BPP_BEAN_NAME)) {
 			registry.registerBeanDefinition(INTEGRATION_FLOW_BPP_BEAN_NAME,
 					new RootBeanDefinition(IntegrationFlowBeanPostProcessor.class));
 			registry.registerBeanDefinition(INTEGRATION_FLOW_CONTEXT_BEAN_NAME,
 					new RootBeanDefinition(StandardIntegrationFlowContext.class));
+		}
+
+		if (!configurableListableBeanFactory.containsBean(INTEGRATION_FLOW_REPLY_PRODUCER_CLEANER_BEAN_NAME)) {
```

Comment:
Wouldn't this be a problem if the child context is loaded by a different class loader?

Reply:
Gary, do you mean that `IntegrationFlowDefinition` is going to be loaded one more time in the child context when it uses its own class loaded?
Then indeed that `private static final Set<MessageProducer> REFERENCED_REPLY_PRODUCERS = new HashSet<>();` store is going to be a fresh in that class loaded...
But I'm not so good in such a low level of JVM.
So, please, confirm how it is going and I'll change respectively or let's leave it as is!

Thanks

Reply:
Yes, that's what I meant; it's unlikely because, normally, class loaders are hierarchical and the child loader will find the existing class instance in the parent loader.

However, it's possible, so the question is whether saving an extra bean in the child context is worth the risk of a memory leak; but I think the risk is very small - weird class loaders combined with a lot of dynamic flow recycling, so I really don't mind either way - I just thought it was worth raising.

Reply:
Just reverted that change.

Thank you for clarification!